relayer/src/prover_interface.rs

```diff
@@ -241,10 +241,18 @@ pub async fn prove_final_with_block_finality(
     previous_proof: ProofWithCircuitData,
     genesis_config: GenesisConfig,
     (block_hash, justification, headers): (H256, GrandpaJustification<GearHeader>, Vec<GearHeader>),
+    inclusion_proof: Option<RawBlockInclusionProof>,
     count_thread: Option<usize>,
 ) -> anyhow::Result<FinalProof> {
     let (_block, block_finality): (H256, dto::BlockFinalityProof) =
-        gear_api.produce_finality_proof(&justification).await?;
+        if let Some(proof) = inclusion_proof {
+            proof.into()
```

Comment:
we need an assert for the case that the last header hash is equal to the block_hash of the proof